src/test/java/core/basesyntax/service/TransactionParserServiceTest.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.impl.TransactionParserServiceImpl;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class TransactionParserServiceTest {
+    private static TransactionParserService parserService;
+
+    @BeforeAll
+    static void beforeAll() {
+        parserService = new TransactionParserServiceImpl();
+    }
+
+    @Test
+    public void getTransactions_validInput_ok() {
+        List<String> validLines = List.of("type,fruit,quantity", "b,apple,100",
+                "b,banana,20", "s,banana,100", "p,banana,13", "r,apple,10",
+                "p,apple,20", "p,banana,5", "r,banana,13");
+        List<FruitTransaction> expected = List
+                .of(new FruitTransaction(FruitTransaction.Operation.BALANCE, "apple", 100),
+                        new FruitTransaction(FruitTransaction.Operation.BALANCE, "banana", 20),
+                        new FruitTransaction(FruitTransaction.Operation.SUPPLY, "banana", 100),
+                        new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 13),
+                        new FruitTransaction(FruitTransaction.Operation.RETURN, "apple", 10),
+                        new FruitTransaction(FruitTransaction.Operation.PURCHASE, "apple", 20),
+                        new FruitTransaction(FruitTransaction.Operation.PURCHASE, "banana", 5),
+                        new FruitTransaction(FruitTransaction.Operation.RETURN, "banana", 13));

```

Comment:
To increase the readability, you can create private helper method like this:
`private FruitTransaction transaction(...)`

And calling it, passing your values. Also, consider using static imports for FruitTransaction.Operation.*.